api_client/client.py

```diff
@@ -95,21 +95,23 @@ def _login(self, conn=None, headers=None):
         return {}
 
     def request(self, opt, content_type=DEFAULT_CONTENT_TYPE,
-                http_timeout=None, **message):
+                http_timeout=None, extra_headers={}, **message):
```

Comment:
can you reuse set_auth_cookie or set_auth_data default in base.py and use wait_for_login workflow to initiate the cookie and save it by self._set_provider_data